File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.js
/**
 * @param {string} s
 * @return {number}
 */
var countPalindromicSubsequence = function(s) {
    const first = Array(26).fill(Infinity)
    const last = Array(26).fill(Infinity)
    for (let i=0; i < s.length; i++) {
        const c = s[i]
        const idx = c.charCodeAt(0)-97
        first[idx] = Math.min(first[idx], i)
        last[idx] = i
    }
    let res = 0
    const set = new Set()
    for (let i=0; i < 26; i++) {
        if (first[i] !== last[i]) {
            set.clear()
            for (let j=first[i]+1; j < last[i]; j++) {
                set.add(s[j])
            }
            res += set.size
        }
    }
    return res
};

File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        first, last, seen, res = [float("inf")] * 26, [float("inf")] * 26, set(), 0
        for i, c in enumerate(s):
            first[ord(c)-97] = min(first[ord(c)-97], i)
            last[ord(c)-97] = i
        for i in range(0, 26):
            if first[i] != last[i]:
                seen = set()
                for j in range(first[i]+1, last[i]):
                    seen.add(s[j])
                res += len(seen)
        return res
        